### PR TITLE
exporters/datadogexporter: update and improve examples

### PR DESCRIPTION
**Pull request by gbbr**
This change deletes the old examples which were no longer working, were
based on deprecated values and used older APIs. It replaces them with
newer up-to-date ones which are tested and known to work.

Added:
* example/collector.yaml: an example configuration yaml for the collector, set up
with Datadog, containing all current options along with their documentation.
* example/k8s-chart: contains a chart which sets up the collector in a deployment,
ready to collect traces and add K8S attribute.

In a follow-up PR, a README.md should be provided to better explain the
purpose of these.

### CODE REVIEW

File: exporter/datadogexporter/examples/collector.yaml
Comment (mx-psi) [inline exporter/datadogexporter/examples/collector.yaml:16]: Don't remove the deprecated settings for now, I will do that on #12110

File: exporter/datadogexporter/examples/sdk-go/go.mod
Comment (mx-psi) [inline exporter/datadogexporter/examples/sdk-go/go.mod:1]: Anything other than Collector YAML configuration samples should not be in this repository (see #7155 and #11074 for precedent)

File: exporter/datadogexporter/examples/k8s-agent/configmap.yaml
Comment (mx-psi) [inline exporter/datadogexporter/examples/k8s-agent/configmap.yaml:33]:
```suggestion
```

**Review by mx-psi — CHANGES_REQUESTED** [review]

File: exporter/datadogexporter/examples/collector.yaml
Comment (gbbr) [inline exporter/datadogexporter/examples/collector.yaml:188]: This option doesn't exist in the code. Not sure why it is here.

File: exporter/datadogexporter/examples/k8s-agent/configmap.yaml
Comment (mx-psi) [reply exporter/datadogexporter/examples/k8s-agent/configmap.yaml:33]: Okay, let's keep it this way then

File: exporter/datadogexporter/examples/k8s-agent/deployment.yaml
Comment (dineshg13) [inline exporter/datadogexporter/examples/k8s-agent/deployment.yaml:18]: Could change the name & labels from `otel-collector` to `app` ?

File: exporter/datadogexporter/examples/collector.yaml
Comment (mx-psi) [inline exporter/datadogexporter/examples/collector.yaml:282]: I would remove this since this is the default
```suggestion
```

Comment (gbbr) [reply exporter/datadogexporter/examples/collector.yaml:282]: Agreed. There is no gain in documenting OTel options. The purpose here should be to underline Datadog exporter config, without any extra noise.

**Review by mx-psi — COMMENTED** [review]
Lint is failing because of a typo; otherwise this LGTM

File: exporter/datadogexporter/examples/k8s-agent/service.yaml
Comment (dineshg13) [inline exporter/datadogexporter/examples/k8s-agent/service.yaml:5]: Could you name this file serviceaccount.yaml ?

Comment (gbbr) [reply exporter/datadogexporter/examples/k8s-agent/service.yaml:5]: Yes! Should I just move it to `roles.yaml`? Is that going to be better? I feel it's perhaps not worth its own file.

Comment (gbbr) [reply exporter/datadogexporter/examples/k8s-agent/service.yaml]: Done!

**Comment by mx-psi** [comment]
Lint is still failing:
```
 running misspell -error
/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/datadogexporter/examples/k8s-chart/deployment.yaml:29:79: "succesfully" is a misspelling of "successfully"
```

**Comment by mx-psi** [comment]
Merging #12110 caused a merge conflict here, could you fix it? Hopefully git is smart enough to make this smooth...

File: exporter/datadogexporter/examples/k8s-chart/configmap.yaml
Comment (dineshg13) [inline exporter/datadogexporter/examples/k8s-chart/configmap.yaml:32]: can u add metrics as well here ?

Comment (gbbr) [reply exporter/datadogexporter/examples/k8s-chart/configmap.yaml:36]: Good point. Added.

**Review by mx-psi — APPROVED** [review]
LGTM (but will wait until @dineshg13 approves as well to merge)

**Review by dineshg13 — APPROVED** [review]
LTGM

**Pull request merged**
